boards/BMS/Src/App/App_Eeprom.c

```diff
@@ -144,3 +143,59 @@ EEPROM_StatusTypeDef App_Eeprom_Read4CopiesOfAddresses(struct Eeprom *eeprom, ui
 
     return read_status;
 }
+
+float App_Eeprom_ReadErrCheckedFloat(struct Eeprom *eeprom, uint16_t address)
+{
+    float floats_read[SAVED_COPIES];
+
+    EEPROM_StatusTypeDef read_status =
+        App_Eeprom_ReadFloats(eeprom, address, DEFAULT_OFFSET, floats_read, SAVED_COPIES);
+
+    // If read unsucessful, reset to 0
+    if (read_status != EEPROM_OK)
+    {
+        return 0.0f;
+    }
+    // If any two read values are the same, return the shared value
+    else if (floats_read[0] == floats_read[1])
```

Comment:
the entire block of if statements could be simplied by a double for loop with `j` starting one past `i`. not sure if that's less clear tho
```suggestion
    for (i = 0; i <= 3; i++)
    {
        for (j = i + 1; j <= 3; j++)
        {
            if (floats_read[i] == floats_read[j]) return floats_read[i];
        }
    }
```